SimMuon/GEMDigitizer/src/GEMDigiProducer.cc

```diff
@@ -39,8 +40,8 @@ GEMDigiProducer::GEMDigiProducer(const edm::ParameterSet& ps)
       << "GEMDigiProducer::GEMDigiProducer() - RandomNumberGeneratorService is not present in configuration file.\n"
       << "Add the service in the configuration file or remove the modules that require it.";
   }
-  gemDigiModel_ = GEMDigiModelFactory::get()->create("GEM" + digiModelString_ + "Model", ps);
-  LogDebug("GEMDigiProducer") << "Using GEM" + digiModelString_ + "Model";
+
+  LogDebug("GEMDigiProducer").log([&](auto& l){ l << "Using GEM" + ps.getParameter<std::string>("digiModelString") + "Model"; });
```

Comment:
what is the reason for using a (seemingly trivial) lambda here?

Reply:
To avoid (for sure) retrieving the `"digiModelString"` and assembling the string unless debug messages are enabled. (ok, the compiler really should do it with regular `operator<<()` as well, and I don't have evidence for otherwise, so I can revert this part as well)

Reply:
I would prefer to revert it

Reply:
Done. (I think it would make sense to squash the update before merging)